src/backend/commands/matview.c

```diff
@@ -1754,6 +1755,9 @@ register_delta_ENRs(ParseState *pstate, Query *query, List *tables)
 
 			nsitem = addRangeTableEntryForENR(pstate, makeRangeVar(NULL, enr->md.name, -1), true);
 			rte = nsitem->p_rte;
+			/* if base table has RLS, set secuirty condition to enr*/
+			rte->securityQuals = get_securityQuals(table->table_id, query);
```

Comment:
Maybe get_securityQuals is not necessary in register_delta_ENRs at least for the current implementation because ENRs are not directly accessed in the primary query rather these are used in subqueries built from SQL strings. However, it will harmless and might be safety.

Reply:
Thank you for your comment.
I think, as safety is important, this is fine.